supernode/httpclient/origin_http_client.go

```diff
@@ -132,18 +132,16 @@ func (client *OriginClient) GetContentLength(url string, headers map[string]stri
 
 // IsSupportRange checks if the source url support partial requests.
 func (client *OriginClient) IsSupportRange(url string, headers map[string]string) (bool, error) {
-	// set headers
-	if headers == nil {
-		headers = make(map[string]string)
-	}
-	headers["Range"] = "bytes=0-0"
+	// set headers: headers is a reference to map, should not change it
+	copied := CopyHeader(nil, headers)
+	copied["Range"] = "bytes=0-0"
```

Comment:
Is this same as `delete(copied, "Range")`?

Reply:
Yeah, but I'm afraid that the `header` maybe read in other places, so I just copy it.

Reply:
> Is this same as `delete(copied, "Range")`?

No, it just read 1 byte from the remote server to test whether the server supports `range` request.

Reply:
Oh, I see. Thanks!